src/onchain.hs

```diff
@@ -111,10 +119,15 @@ mintValidate c action ctx = case action of
                                                   -- | Correct lock output datum
                                                   (PlutusTx.unsafeFromBuiltinData lockOutDatum :: Api.CurrencySymbol) == ownSymbol &&
                                                   -- | Check if metadata and asset names belong together and are part of the merkle tree
+                                                  -- REVIEW(SN): How big are your merkle tree proofs?
+                                                  -- I read somewhere you would have 10000 entries in the tree.
                                                   MT.member merkleEntry (merkleRoot c) merkleProof &&
                                                   -- | Loop again 
                                                   checkHelper merkleProofT refOutT lockOutT (refL - 1) (userL - 1)
                                     mint = V.flattenValue txMint
+                                    -- REVIEW(SN): Isn't it relevant that the refAddres/lockAddress correspond to the
+                                    -- referenceValidate/lockValidate validators below? How would the minting policy know
+                                    -- that these rules are enforced?
```

Comment:
`refAddress` and `lockAddress` are parameters of the minting policy. They do refer to `referenceValidate` and `lockValidate`. You can certainly verify off-chain that the minting policy was compiled with the right addresses and that they do belong to the correct validators. I don't know what you mean with how the minting policy can enforce these rules?

Reply:
> You can certainly verify off-chain that the minting policy was compiled with the right addresses and that they do belong to the correct validators.

That was the link I was missing, but needs to be also done. It could be evident from the plutus-tx code itself if you would use `applyCode` and `liftCode` with the `scriptAddress` of the two validators. That way, the resulting blob is not needing to be correctly parameterized for the other two validators. That could make sense if you intend to refer to the minting policy always together with the appropriate validators.